src/Microsoft.Azure.WebJobs.Host/Executors/DefaultStorageCredentialsValidator.cs

```diff
@@ -71,21 +69,30 @@ internal class DefaultStorageCredentialsValidator : IStorageCredentialsValidator
             {
                 await queue.ExistsAsync(cancellationToken);
             }
-            catch (OperationCanceledException)
+            catch (StorageException exception) when (IsBlobOnlyStorageException(exception))
             {
-                throw;
+                account.Type = StorageAccountType.BlobOnly;
             }
-            catch (StorageException exception)
+        }
+
+        private static bool IsBlobOnlyStorageException(StorageException storageException)
+        {
+            const int errorWinhttpNameNotResolved = 0x2ee7;
+            const int serviceUnavailable = 503;
+
+            switch (storageException.InnerException)
             {
-                WebException webException = exception.GetBaseException() as WebException;
-                if (webException?.Status == WebExceptionStatus.NameResolutionFailure)
-                {
-                    // Blob-only storage accounts do not support services other than Blob.  
-                    // If we see a name resolution failure on the queue endpoint classify as a blob-only account
-                    account.Type = StorageAccountType.BlobOnly;
-                    return;
-                }
-                throw;
+                // Inner exception when using WindowsAzure.Storage for net45
+                case WebException webException:
+                    return webException.Status == WebExceptionStatus.NameResolutionFailure;
+                // Inner exception when using WindowsAzure.Storage for netstandard1.3 as WebException does not exist
+                case System.Net.Http.HttpRequestException httpRequestException:
+                    return (httpRequestException.HResult & 0xFFFF) == errorWinhttpNameNotResolved;
+                // When running behind a proxy
+                case null:
+                    return storageException.RequestInformation?.HttpStatusCode == serviceUnavailable;
```

Comment:
Can you explain this case? Inferring a blob-only storage account from something as generic as a 503 doesn't seem right but perhaps I'm missing something.

Reply:
In fact, I've tried this code at home and at work. At work, we're behind a corporate http proxy (squid) which "catch" the resolution error and gives a 503. Not sure if it's specific to squid or all the proxy. Maybe we can go without ?

Reply:
@NatMarchand This doesn't seem resolved. I still have issues with inferring a blob-only account from a generic 503...

Reply:
Ok, will remove the case when rebasing then :)